### PR TITLE
[docs] docs: Clarify Quick Start model source workflows

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary

- Clarifies that HF -> Megatron checkpoint conversion is workflow-specific, not required for random-init pretraining from an HF architecture.
- Shows the pre-configured recipe pattern for model sources: HF model ID, local HF path, and converted Megatron checkpoint initialization.
- Links the repository tutorials directory from the Quick Start path so users can find runnable data prep, recipe, and training examples earlier.
- Normalizes Llama tutorial dataset override examples so pretraining uses `dataset.data_path` and finetuning uses `dataset.dataset_root`.

## Scope

Addresses the Quickstart-work subset of #3814 and keeps the tutorial examples aligned with the config-name normalization covered by the training-docs PR. This PR does not cover the rest of the tracker items such as training workflow docs, NVFP4, checkpoint resume semantics, data prep guide expansion, or agent readiness.

## Verification

- `git diff --check`
- `pre-commit run --files README.md docs/README.md tutorials/recipes/llama/README.md`
- `pre-commit run --files tutorials/recipes/llama/00_quickstart_pretrain.py tutorials/recipes/llama/README.md tutorials/recipes/llama/conf/llama32_1b_finetune.yaml`
- `pre-commit run --all-files`
- Attempted `uv run pre-commit run --all-files`; blocked by local dependency resolution for `nvidia-resiliency-ext==0.6.0` on `manylinux_2_31_x86_64` before hooks ran.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,14 +129,27 @@ for name, weight in bridge.export_hf_weights(model, cpu=True):
...
 
 if __name__ == "__main__":
-    # The recipe uses the Llama 3.2 1B model configuration from HuggingFace
+    # The recipe uses the Llama 3.2 1B architecture from Hugging Face.
+    # This is random-init pretraining and does not require a converted Megatron checkpoint.
     cfg = llama32_1b_pretrain_config()
 
+    # Hugging Face model ID as the architecture source.
+    # The recipe uses this pattern internally; override cfg.model to choose a different source.
+    cfg.model = AutoBridge.from_hf_pretrained("meta-llama/Llama-3.2-1B").to_megatron_provider(load_weights=False)
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply README.md]: Fixed in 92d4458a30d555c63a8be39b83eee372e9ce1a5f. The HF model ID example is now commented out, so the runnable snippet keeps the provider settings applied by llama32_1b_pretrain_config().

**Comment by yaoyu-33** [comment]
Addressed the review comments in 92d4458a30d555c63a8be39b83eee372e9ce1a5f: the cfg.model override is commented so the runnable snippet preserves recipe defaults. Applied labels docs, docs-only, and area:recipe. The inline review thread is resolved.

**Review by cuichenx — APPROVED** [review]
